js/ui/animation.js

```diff
@@ -91,6 +100,14 @@ var Animation = new Lang.Class({
     _animationsLoaded: function() {
         this._isLoaded = this._animations.get_n_children() > 0;
 
+        if (this._isLoaded && this._frames.length === 0) {
+            // If a custom sequence of frames wasn't provided,
+            // fallback to play the frames in sequence.
+            for (let i = 0; i < this._animations.get_n_children(); i++) {
+                this._frames.push({'frame': i, 'delay': this._speed});
+            }
```

Comment:
Can be rewritten as `this._frames = new Array(this._animations.get_n_children()).fill({'frame': i, 'delay': this._speed})`

Reply:
No, I couldn't use `fill` here.

Reply:
I think you meant `map` instead of `fill`.